### PR TITLE
Fix : LT-18668 Crash creating new project - Migration of WS Fails

### PR DESCRIPTION
**Pull request by samdoss**
* Fill the value in WS language id.

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/576)

### CODE REVIEW

**Comment by papeh** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[SIL.WritingSystems/GlobalWritingSystemRepository.cs, line 104 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/576#-Kw0pI2hGFLQ7NZsW3oc:-Kw0pI2iszg0xNSKhQPJ:b-n11lhw) ([raw file](https://github.com/sillsdev/libpalaso/blob/435e8814c829969cc5a708089ce10c5933562825/SIL.WritingSystems/GlobalWritingSystemRepository.cs#L104)):*
> ```Smalltalk
> 							ldmlDataMapper.Read(file, ws);
> 							if(string.IsNullOrEmpty(ws.Id))
> 								ws.Id = ws.LanguageTag;
> ```

This will work as long as no other WS's have the same LanguageTag

---

*[SIL.WritingSystems.Tests/GlobalWritingSystemRepositoryTests.cs, line 308 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/576#-Kw0oSWvmrBuASWSMYUN:-Kw0oSWwcJnfJiuQrPBT:benslln) ([raw file](https://github.com/sillsdev/libpalaso/blob/435e8814c829969cc5a708089ce10c5933562825/SIL.WritingSystems.Tests/GlobalWritingSystemRepositoryTests.cs#L308)):*
> ```Smalltalk
> 		public void AllWritingSystems_LdmlCheckingSetEmptyandGetWSId()
> 		{
> 			using (var e = new TemporaryFolder("GlobalWritingSystemRepositoryTests"))
> ```

`using (var tf` would be slightly more descriptive

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/576)*

File: SIL.WritingSystems.Tests/GlobalWritingSystemRepositoryTests.cs [hunk 1]
```diff
@@ -301,5 +301,29 @@ public void Count_LdmlRemovedByAnotherRepo_ReturnsZero()
...
+				var repo2 = new GlobalWritingSystemRepository(e.Path);
+
+				var ws = new WritingSystemDefinition("en-US");
+				repo1.Set(ws);
+				repo1.Save();
+				Assert.That(ws.WindowsLcid, Is.Empty);
+				// ensure that last modified timestamp changes
+				Thread.Sleep(1000);
+				ws = repo2.Get("en-US");
+				ws.WindowsLcid = "test";
```
Comment (jasonleenaylor) [inline SIL.WritingSystems.Tests/GlobalWritingSystemRepositoryTests.cs]: Please add a comment here:
"Make an arbitrary change to force a save of the ldml file."

**Comment by papeh** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[SIL.WritingSystems/GlobalWritingSystemRepository.cs, line 104 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/576#-Kw0pI2hGFLQ7NZsW3oc:-Kw0ufpQfR4go5qv6iVV:b5qhgtt) ([raw file](https://github.com/sillsdev/libpalaso/blob/435e8814c829969cc5a708089ce10c5933562825/SIL.WritingSystems/GlobalWritingSystemRepository.cs#L104)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

This will work as long as no other WS's have the same LanguageTag
</blockquote></details>

But we do this in other places without checking, so don't worry about it. :LGTM: as is.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/576)*

File: SIL.WritingSystems.Tests/GlobalWritingSystemRepositoryTests.cs (hunk 1 above)
Comment (samdoss) [reply SIL.WritingSystems.Tests/GlobalWritingSystemRepositoryTests.cs]: Added the commen in the GlobalWritingSystemRepositoryTest.cs file.

**Comment by samdoss** [comment]
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[SIL.WritingSystems.Tests/GlobalWritingSystemRepositoryTests.cs, line 308 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/576#-Kw0oSWvmrBuASWSMYUN:-Kw42gmP-6GB7W2tl-7a:byeusjo) ([raw file](https://github.com/sillsdev/libpalaso/blob/435e8814c829969cc5a708089ce10c5933562825/SIL.WritingSystems.Tests/GlobalWritingSystemRepositoryTests.cs#L308)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

`using (var tf` would be slightly more descriptive
</blockquote></details>

Changed passed on your suggestion. Thank you.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/576)*

**Pull request merged**